common/src/main/scala/org/apache/celeborn/common/rpc/netty/Inbox.scala

```diff
@@ -278,6 +278,11 @@ private[celeborn] class Inbox(
         // We should disable concurrent here. Then when RpcEndpoint.onStop is called, it's the only
         // thread that is processing messages. So `RpcEndpoint.onStop` can release its resources
         // safely.
+        // we need to have this check in case that inbox has been full and we have no way to
+        // put OnStop leading the deadlock when shutting down
+        if (capacity > 0 && messageCount.get() >= capacity) {
```

Comment:
Why does this add this check? IMO, we could directly clear the messages.

Reply:
it is to ensure that we only clear the messages when the queue is full, additionally, it is to pass ut

Reply:
@CodingCat, the source of truth should be clear the messages regardless of whether the queue is full or not for `stop`.

Reply:
> @CodingCat, the source of truth should be that clearing the messages regardless of whether the queue is full or not for `stop`. We do not need to add this check for passing ut.

The current code doesn't seem to have any obvious issues, and I prefer to keep the status, so I will close this PR unless new problems arise. cc @SteNicholas @CodingCat